app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListFragment.kt

```diff
@@ -164,6 +163,23 @@ class ThreadListFragment : Fragment(), SwipeRefreshLayout.OnRefreshListener {
         observeUpdateInstall()
     }.getOrDefault(Unit)
 
+    private fun navigateFromNotificationToThread() {
+        threadListObserver = object : AdapterDataObserver() {
+            override fun onChanged() {
+                // Here, we use `arguments` instead of `navigationArgs` because we need mutable data.
+                if (arguments?.getString(navigationArgs::openThreadUid.name) != null) {
+                    navigationArgs.openThreadUid?.let { openThreadUid ->
+                        // If we are coming from a Notification, we need to navigate to ThreadFragment.
+                        navigateToThread(mainViewModel, threadUid = openThreadUid)
+                        arguments?.remove(navigationArgs::openThreadUid.name)
+                        threadListObserver?.let(threadListAdapter::unregisterAdapterDataObserver)
+                    }
+                }
+            }
+        }
+        threadListObserver?.let(threadListAdapter::registerAdapterDataObserver)
+    }
+
```

Comment:
Optimize the opening of notifications like the commit in `opti-opening-notifications` does so we don't wait on the adapter for no reason. Probably best to put it in its own PR though

Reply:
Fixed in https://github.com/Infomaniak/android-kMail/pull/1606?w=1